### ISSUE TITLE
RFE: Have file attributes for %license and %doc

### ISSUE THREAD

**Issue by ffesti** [rpm-software-management/rpm#713]
Right now files need to be marked in the file list as doc or license. We already have the fileattr patterns to categorise files (for build dependency generators). We could either use some file attrs to mark files as doc and license or have some generators that even filers the file lists.

**Comment by pmatilai** [comment]
Yeah, emitting virtual file flags from dependency generator could be a useful thing, %artifact is yet another such thing. Additionally file coloring could be outsourced to the generators, that's where it really belongs. This just needs a new API for the generator scripts, but this could be a per-fileattr switchable - by default they use the current "API", but allow them to opt-in to a version that expects colors and file attributes (and what else?) from the generator output too.

**Comment by hroncok** [comment]
I think that for most common cases, [people don't want dependency generators for `%doc` and `%license`](https://bugzilla.redhat.com/show_bug.cgi?id=964126). Hence it might make sense to not process such files by default at all and add an attr flag that can enable to process them, such as:

```
%__NAME_flags doc license
```

That would still *not* expose the attributes to the generator but OTOH it doesn't need a new API.

**Comment by fundawang** [comment]
There is another use case for mageia. They are shipping META.json/META.yml for perl-* packages in %{_docdir}, then scanning it for dependencies. META.json/META.yml are of no use for the package itself, the only use is to generating dependencies.

Currently, magiea is commenting out these lines:
https://svnweb.mageia.org/packages/cauldron/rpm/current/SOURCES/dont-filter-autodeps-from-doc-by-default.patch
